src/index.js

```diff
@@ -167,26 +168,32 @@ export default class TimeField extends React.Component {
   }
 
   render() {
-    const {value} = this.state;
-    const {onChange, style, showSeconds, input, colon, ...props} = this.props; //eslint-disable-line no-unused-vars
-    const onChangeHandler = (event) => this.onInputChange(event, (v) => onChange(v));
+    const { value } = this.state;
+    const { onChange, style, showSeconds, input, inputRef, colon, defaultValue, ...props } = this.props; //eslint-disable-line no-unused-vars
+    const onChangeHandler = (event) => this.onInputChange(event, (v) => onChange(event, v));
 
     if (input) {
       return React.cloneElement(input, {
         ...props,
+        colon,
         value,
         style,
-        onChange: onChangeHandler
+        defaultValue,
+        ref: inputRef,
```

Comment:
I can imagine the situation where you need to pass `inputRef` to the cloned element.
I'm trying to figure out can the same problem be solved by replacing this code:
```javascript
<TimeField input={<MyInput />} inputRef={myRef} />
```
By this one:
```javascript
<TimeField input={<MyInput ref={myRef} />} />
```
?

Reply:
inputRef need only native DOM input, maybe i don't  understand your code